src/model/BoxPlotColumn.ts

```diff
@@ -0,0 +1,141 @@
+/**
+ * Created by bikramkawan on 24/11/2016.
+ */
+import * as d3 from 'd3';
+import ValueColumn from './ValueColumn';
+import Column from './Column';
+import {IValueColumnDesc} from './ValueColumn';
+import {numSort} from './MultiValueColumn';
+
+export enum Sort {
+
+  min, max, median, q1, q3, mean
+}
+
+
+export interface IBoxPlotColumn {
+  getBoxPlotData(row: any, index: number): IBoxPlotData;
+  getDataInfo(): IBoxPlotColumnDesc;
+  getUserSortBy(): string;
+  setUserSortBy(sortCriteria: string): void;
+
+}
+
+/*
+ This was created only for targid purpose to set the domain I
+
+ interface IBoxPlotDataStat extends IDataStat {
+ min: number,
+ max: number,
+ readonly sort: string
+
+ }
+
+ */
+
+export interface IBoxPlotColumnDesc extends IValueColumnDesc < IBoxPlotData > {
+
+  readonly domain?: number [];
+  readonly sort?: string;
+  readonly colorRange?: any;
+  readonly min?: number;
+  readonly max?: number;
+
+}
+
+export  interface IBoxPlotData {
+
+  min: number;
```

Comment:
can't you just not create a new object instead of updating the old one, would be more clear to have the fields readonly